pkg/services/user/userimpl/user.go

```diff
@@ -594,3 +598,25 @@ func (s *Service) supportBundleCollector() supportbundles.Collector {
 		Fn:                collectorFn,
 	}
 }
+
+func hashUserIdentifier(identifier string, secret string) string {
+	key := []byte(secret)
+	h := hmac.New(sha256.New, key)
+	h.Write([]byte(identifier))
+	return hex.EncodeToString(h.Sum(nil))
+}
```

Comment:
taken from https://gist.github.com/thewheat/7342c76ade46e7322c3e#go (linked from the [intercom docs](https://www.intercom.com/help/en/articles/183-enable-identity-verification-for-web-and-mobile))